jax/lax_linalg.py

```diff
@@ -872,11 +884,14 @@ def svd_abstract_eval(operand, full_matrices, compute_uv):
     m = operand.shape[-2]
     n = operand.shape[-1]
     s = ShapedArray(batch_dims + (min(m, n),), lax.lax._complex_basetype(operand.dtype))
-    u = ShapedArray(batch_dims + (m, m if full_matrices else min(m, n)), operand.dtype)
-    vt = ShapedArray(batch_dims + (n if full_matrices else min(m, n), n), operand.dtype)
+    if compute_uv:
+      u = ShapedArray(batch_dims + (m, m if full_matrices else min(m, n)), operand.dtype)
+      vt = ShapedArray(batch_dims + (n if full_matrices else min(m, n), n), operand.dtype)
+      return s, u, vt
+    else:
+      return s,
   else:
     raise NotImplementedError
-  return s, u, vt
 
 def svd_jvp_rule(primals, tangents, full_matrices, compute_uv):
```

Comment:
I think you need to fix the JVP rule also? The output signature of the JVP rule should mirror that of the regular primitive.

Reply:
Addressed in last commit.
Yes, there is a in `lax/linalg_test.py`, which somehow is not failing. I am not entirely sure why yet?

Reply:
One thing that is always worth checking is that we aren't failing to sample such a test. We only run a random sample of possible tests. Try setting `JAX_NUM_GENERATED_CASES=100` (or larger) and running that test case.

Reply:
The branch is definitely hit, but there is no reported failure with or without the fix to the jvp rule, it is odd.

Reply:
This had to do with the way I unpacked the value in `svd`. I fixed it in the last commit, this is resolved.